components/templates/src/global_fns/load_data.rs

```diff
@@ -23,6 +24,26 @@ enum DataSource {
     Path(PathBuf),
 }
 
+#[derive(Debug, PartialEq, Clone, Copy, Hash)]
+enum Method {
+    Post,
+    Get,
+}
+
+impl FromStr for Method {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self> {
+        if s.to_lowercase() == "post" {
+            Ok(Method::Post)
+        } else if s.to_lowercase() == "get" {
+            Ok(Method::Get)
+        } else {
+            return Err("`load_data` method must either be POST or GET.".into());
+        }
+    }
```

Comment:
```rust
match s.lowercase() {
  "post" => Ok(Method::Post),
  "get" => Ok(Method::Get),
  _ => Err("`load_data` method must either be POST or GET.".into())
}
```
would be more idiomatic

Reply:
just commited the change to make the suggested change